File: src/main/java/com/fifo/ticketing/domain/user/service/handler/FormLoginFailureHandler.java
package com.fifo.ticketing.domain.user.service.handler;

import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.IOException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.authentication.AuthenticationFailureHandler;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class FormLoginFailureHandler implements AuthenticationFailureHandler {

    @Override
    public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response,
        AuthenticationException exception) throws IOException, ServletException {
        String errorMessage = "이메일 또는 비밀번호가 올바르지 않습니다.";

        request.getSession().setAttribute("errormessage", errorMessage);
        response.sendRedirect("/users/signin");
    }
}

File: src/main/java/com/fifo/ticketing/domain/user/service/handler/FormLoginFailureHandler.kt
package com.fifo.ticketing.domain.user.service.handler

import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.authentication.AuthenticationFailureHandler
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class FormLoginFailureHandler : AuthenticationFailureHandler {

    @Throws(IOException::class, ServletException::class)
    override fun onAuthenticationFailure(
        request: HttpServletRequest,
        response: HttpServletResponse,
        exception: AuthenticationException
    ) {
        val errorMessage = "이메일 또는 비밀번호가 올바르지 않습니다."

        request.session.setAttribute("errormessage", errorMessage)
        response.sendRedirect("/users/signin")
    }
}
